### PR TITLE
Use new LINQ Order() methods instead of OrderBy(static x => x)

### PR DESCRIPTION
**Pull request by turbedi**
# PR Summary

Less code, a bit faster, less allocations.
Introduced in .NET 7 Preview 7

Docs for [Order<T>(IEnumerable<T>)](https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.order?view=net-7.0#system-linq-enumerable-order-1(system-collections-generic-ienumerable((-0)))) and [Order<T>(IEnumerable<T>, IComparer<T>)](https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.order?view=net-7.0#system-linq-enumerable-order-1(system-collections-generic-ienumerable((-0))-system-collections-generic-icomparer((-0))))

Related dotnet/runtime [PR](https://github.com/dotnet/runtime/pull/70525) and [issue](https://github.com/dotnet/runtime/issues/67194): 
## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Comment by turbedi** [comment]
@microsoft-github-policy-service agree

**Review by iSazonov — APPROVED** [review]

**Comment by xtqqczze** [comment]
@turbedi Please link to documentation for `Order<T>(IEnumerable<T>)` in the pull request description.

**Comment by turbedi** [comment]
Added links to documentation to the related dotnet/runtime PR and issue.

**Comment by xtqqczze** [comment]
@iSazonov Should we be caching `IComparer<PSObject>` objects, e.g. `CommandNameComparer`?

**Comment by xtqqczze** [comment]
@xtqqczze We should investigate whether `List<CommandInfo> _accumulatedResults` can be of type `IEnumerable<CommandInfo>`, in order to avoid allocations.

File: src/System.Management.Automation/CoreCLR/CorePsAssemblyLoadContext.cs [hunk 1]
```diff
@@ -401,7 +401,7 @@ private static bool FindInGac(string gacRoot, AssemblyName assemblyName, out str
             if (Directory.Exists(tempAssemblyDirPath))
             {
                 // Enumerate all directories, sort by name and select the last. This selects the latest version.
-                var chosenVersionDirectory = Directory.EnumerateDirectories(tempAssemblyDirPath).OrderBy(static d => d).LastOrDefault();
+                var chosenVersionDirectory = Directory.EnumerateDirectories(tempAssemblyDirPath).Order().LastOrDefault();
```
[the remark this answers is not in the document]
Comment (turbedi) [reply src/System.Management.Automation/CoreCLR/CorePsAssemblyLoadContext.cs]: Replaced the OrderBy().LastOrDefault() call with Max()

**Review by xtqqczze — CHANGES_REQUESTED** [review]

**Comment by iSazonov** [comment]
@turbedi @xtqqczze I prefer to keep the PR simple and leave only the first commit.

**Comment by turbedi** [comment]
> @turbedi @xtqqczze I prefer to keep the PR simple and leave only the first commit.

Done

**Comment by xtqqczze** [comment]
@iSazonov @turbedi Could you please push a commit addressing rewording of code comments.

**Comment by iSazonov** [comment]
> @iSazonov @turbedi Could you please push a commit addressing rewording of code comments.

I don't see wrong comments, but you could try to reword them in follow PR.

**Comment by turbedi** [comment]
I can do a follow-up PR to apply the code suggestions and to rename/remove related comments.

**Comment by iSazonov** [comment]
@daxian-dbw CommandPredictionTests doesn't work as expected 😕

**Pull request merged**